### ISSUE TITLE
Broken Z->Y transform conversions & Bone related transforms

### ISSUE THREAD

**Issue by franklinblanco** [kaosat-dev/Blender_bevy_components_workflow#162]
Hey, I've been testing out the new released gltf_auto_export & all the blender_bevy workflow except for the save_load crate.

Everything was going alright, but for some reason, after saving in blender, all the files in the library got overwritten and now the way that things are transformed in relation to bones is all changed. 

I'll show you what I mean:

This is what I have in blender. The selected item is the camera.

![Screenshot 2024-03-05 at 11 06 23 AM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/4ddf2705-377b-49d7-a69f-61da40bec1c2)

You can see it's z location is 1.731 & its parent is the Head bone.
![Screenshot 2024-03-05 at 11 07 05 AM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/bddfab9e-1e30-4375-8b6e-4ce168a5741e)

Here is the character imported into bevy. Note that it's Z translation is the same as blender's Z translation
![Screenshot 2024-03-05 at 11 09 52 AM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/ee64c373-a0d7-4029-8555-36f5ae5d959d)

Even though I have the settings set to +Y up
![Screenshot 2024-03-05 at 11 11 02 AM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/f7baa28d-f37a-4b6a-8a58-2151793b7a42)

Even if the +Y up was fixed, there's still an issue. In blender, anything that's parented to a bone, if put on (0,0,0) would be at the origin of the armature. This got translated to bevy in a very good way, I'm not sure how but it ended up without the translation values in blender but in the same spot as it was in blender.

But now in bevy, it doesn't get translated very well...

![Screenshot 2024-03-05 at 11 13 26 AM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/53fc59a8-c420-44fc-9627-f784b24363b6)

If I git reset the whole library to where it was before this change happened, I get just what I expected to happen. Using the same version of bevy_gltf_components & blueprints.
![Screenshot 2024-03-05 at 11 19 48 AM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/b77acbca-4546-4891-b010-180747f8dfc3)

So I assume it's a problem with the auto_save plugin, I'll try reverting back to an earlier version and see if it fixes the issue

```toml
bevy_gltf_blueprints = { version = "0.9" }
bevy_gltf_components = { version = "0.5" }
```

**Comment by franklinblanco** [comment]
I was using the 0.15.0 auto-export

**Comment by franklinblanco** [comment]
Seems to be fixed when downgrading to auto-export v0.14 for now.

**Comment by franklinblanco** [comment]
Nevermind still not fixed in 0.14, when I added a new asset the same thing happened. Going down to 0.13

**Comment by franklinblanco** [comment]
Happened in 0.13 too. I'm starting to wonder, am I doing something wrong?

**Comment by kaosat-dev** [comment]
Hi @franklinblanco !
I must confess that I am out of my depth with all things animation & co, but i'll try my best to help you:
Did you try with 0.13 or 0.13.1 ? because I change the internals of how hierarchies are copied in v0.13/v0.13.1
It would be a good idea to try it out with 0.12, because the code was very different for hierarchies & animated objects in that version.
If that solves it , it is both good & bad :D

**Comment by franklinblanco** [comment]
>Hi
Hello! Good to speak with ya again!

> I must confess that I am out of my depth with all things animation & co, but i'll try my best to help you:

So it shouldn't be anything to do with animation, as it's just the armature unmoved.

I'll give it a shot with 0.12!

**Comment by kaosat-dev** [comment]
hi @franklinblanco !
I understand your frustration, but we'll figure something out :)
to double check:
* the issue is the same in auto_export 0.12 ?
* did you try exporting the model using the standard gltf exporter (not auto export) and check if it still has the same issues ?
* if the above fails, could you please send me the glb so I can take a look ?

**Comment by franklinblanco** [comment]
> hi @franklinblanco ! I understand your frustration, but we'll figure something out :) to double check:
Thanks so much. And again, excuse the time being taken away.
> * the issue is the same in auto_export 0.12 ?
Correct.
> * did you try exporting the model using the standard gltf exporter (not auto export) and check if it still has the same issues ?
Will do right now.

**Comment by franklinblanco** [comment]
@kaosat-dev Worked as a normal export...
But, just found out, it's not an armature problem, other regular hierarchies are also presenting the same problem and those don't have armatures. I kind of feel very inclined to dig into the gltf_auto_export source code. Even though I hate python I might have to. Any advice on how to start?

**Comment by kaosat-dev** [comment]
@franklinblanco  thanks for trying that out !
but wait wait wait, 
> regular hierarchies are also presenting the same problem
That is very helpful and important information ! It will help a lot to track the issue down.
Can you please give me a bit more detail about 
 * what you mean by "normal hierarchies" (parent/child I guess ?)
 * what happens exactly to those hierarchies (blender vs bevy)
 
 For Python/ etc help, I'll try and give you pointers later ;)

**Comment by franklinblanco** [comment]
Perfect. I'll drop everything I can here:

What:
I have a model in blender of a naked firearm, an AK105. I have various "attachment slots" as a child of the gun entity.
Let's take the handguard slot for example.
Take a look at where it is in blender & specifically the translations.
![Screenshot 2024-03-07 at 1 24 29 PM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/83dab0f1-8cd9-42f3-a8ea-127d2dae2699)
## In bevy (Before this bug started happening), you can see the gun filled with the attachments placed inisde each respective slot looks normal, as should be.
![Screenshot 2024-03-07 at 1 26 08 PM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/1cc08de8-86ea-47ec-a20f-b3ff6575a51e)

## In bevy (Just saved using gltf_auto_export 15.0 & re-run)
Now, you can see, none of the attachments show up, even when you can clearly see in the hierarchy how they're there. Just not where they should be.
![Screenshot 2024-03-07 at 1 41 05 PM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/bb8c5aba-bca0-4b28-b53a-247a6e429655)

As a matter of fact, take a look. Going inside the gun, shows me that the parts, seemingly at random, are inside the gun? 
![Screenshot 2024-03-07 at 1 43 49 PM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/bfaa6a97-3200-414c-893f-4a5dedde99cf)

**Comment by franklinblanco** [comment]
@kaosat-dev Maybe tell me how to do a complete wipe of all the gltf_auto_export data and see if that works?

**Comment by franklinblanco** [comment]
Oh wow. I might have just found another issue. Orphan data is created on every save.

## Auto export details
![Screenshot 2024-03-07 at 2 04 15 PM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/3faa2a26-ef0e-4703-b69d-ddfe8a908cc5)

## Orphan data panel (Just purged it)
![Screenshot 2024-03-07 at 2 04 41 PM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/71bd6786-7485-40b1-a193-54283a36ad10)

## Saved once.
![Screenshot 2024-03-07 at 2 05 13 PM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/94697ccd-69a4-4580-96c4-4765e3cef9d4)


## Saved 3 more times
![Screenshot 2024-03-07 at 2 05 42 PM](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/72827992/2c52dfae-aa37-4474-ab3a-2530439400ba)

**Comment by kaosat-dev** [comment]
@franklinblanco I'll answer in reverse:
* Orphan Data: was solved quite a few versions ago , but apparently, you are right, the latest version has a regression, please open an issue for that one
* wipe of all the gltf_auto_export : it barely stores anything at all , you can delete these two items to clear data: (not the one from bevy_components)
![image](https://github.com/kaosat-dev/Blender_bevy_components_workflow/assets/623281/814aaaf4-96e2-45a8-8c27-386146f93138)
I'll answer the rest later

**Comment by franklinblanco** [comment]
@kaosat-dev Created #167 For the orphan data stuff.
Same result after wiping the data.

I'll keep trying to find the issue, thanks for the time!

**Comment by kaosat-dev** [comment]
@franklinblanco yeah the settings would not have much of an impact.
What is not normal though is that it fails the same way in both auto_export v0.12 & v0.15, as they exported hierarchies in a completely different manner.

Thanks for the screenshots & infos.
The best way to solve this would be minimal / VERY barebones example where it happens
Would you mind creating a small blend file with a simple hierarchy to see if it happens ?
Because in the "testing" folder I already have a minimal one to check for regressions, and it has an object hierarchy, that **seemed** fine to me , but I might have missed something

**Comment by franklinblanco** [comment]
Hey again, Well... After really extensive testing, I still can't hit the nail on why the gltf_auto_export differs from the normal gltf export. But what I did find out is that the glitch seemed to be blender related, as when I switch some of the transforms (seemingly random ones) to use the armature instead of the bone as a parent, then quickly switch them back, the transforms are just wrong. Also I had some scale issues where there weren't any armatures. I will be therefore closing this issue until I get a smaller scale replica made!

Thanks for everything you've done, I'm waiting on getting some more savings so I can start donating some cash, as this really helps me out so much, and you constantly improve upon this all the time. the blender components add on is pure gold.

**Comment by kaosat-dev** [comment]
Hi @franklinblanco !
I am currently in the process of improving all things related to animations & fixing bugs (a lot of which you reported :) , so you should see some improvements soon !
That issue you describe still seems very confusing tbh !
Btw I wanted to ask if you are on discord, so that we can exchange some information faster to reproduce the issue ! 

And hey don't stress out please, I created this tooling for all to use freely and for fun, you shouldn't have to donate anything ! If you find it useful to your needs it really is all that counts :)

**Comment by franklinblanco** [comment]
I just texted abcdefg in the chat for your crate!
